cohorts/load.py

```diff
@@ -489,38 +488,53 @@ def load_variants(self, patients=None, variant_type="snv", merge_type="union"):
 
     def _load_single_patient_variants(self, patient, variant_type, merge_type):
         failed_io = False
-        combined_variants = []
         try:
             cached_file_name = "%s-%s-variants.pkl" % (variant_type, merge_type)
             cached = self.load_from_cache(self.cache_names["variant"], patient.id, cached_file_name)
             if cached is not None:
                 return cached
 
             vcf_paths = patient.snv_vcf_paths if variant_type == "snv" else patient.indel_vcf_paths
-            for vcf_path in vcf_paths:
-                variants = varcode.load_vcf_fast(vcf_path)
-                combined_variants.append(set(variants.elements))
+            variant_collections = [
+                (vcf_path, varcode.load_vcf_fast(vcf_path)) 
+                for vcf_path in vcf_paths
+            ]
         except IOError:
             failed_io = True
 
-        if len(combined_variants) == 0 or failed_io:
+        if len(variant_collections) == 0 or failed_io:
             print("Variants did not exist for patient %s" % patient.id)
             return VariantCollection([])
 
-        if len(combined_variants) == 1:
+        if len(variant_collections) == 1:
             # There is nothing to merge
-            merged_variants =  VariantCollection(combined_variants[0])
+            merged_variants = VariantCollection(variant_collections[0][1].elements)
```

Comment:
@tavinathanson I updated this to actually keep the metadata in the single vcf case as well